ext/AbstractDifferentiationEnzymeExt.jl

```diff
@@ -0,0 +1,60 @@
+module AbstractDifferentiationEnzymeExt
+
+if isdefined(Base, :get_extension)
+    import AbstractDifferentiation as AD
+    using Enzyme: Enzyme
+else
+    import ..AbstractDifferentiation as AD
+    using ..Enzyme: Enzyme
+end
+
+AD.@primitive function jacobian(b::AD.EnzymeForwardBackend, f, x)
+    val = f(x)
+    if val isa Real
+        return adjoint.(AD.gradient(b, f, x))
+    else
+        if length(x) == 1 && length(val) == 1
+            # Enzyme.jacobian returns a vector of length 1 in this case
+            return (Matrix(adjoint(Enzyme.jacobian(Enzyme.Forward, f, x))),)
+        else
+            return (Enzyme.jacobian(Enzyme.Forward, f, x),)
+        end
+    end
+end
+function AD.jacobian(b::AD.EnzymeForwardBackend, f, x::Real)
+    return AD.derivative(b, f, x)
+end
+function AD.gradient(::AD.EnzymeForwardBackend, f, x::AbstractArray)
+    # Enzyme.gradient with Forward returns a tuple of the same length as the input
+    return ([Enzyme.gradient(Enzyme.Forward, f, x)...],)
```

Comment:
Also here

Reply:
why? I am calling `Enzyme.gradient` which in turn calls `autodiff` internall.

Reply:
gradient assumes a function which takes a densearray, not an abstract array, which requires corresponding structural zero initialization.

Reply:
the new implementation uses autodiff